### PR TITLE
Rework/Cleanup the main Discord-Penny target

### PR DESCRIPTION
**Pull request by MahdiBM**
More cleanup. Would specially be nice for better testability when it's done.

### CODE REVIEW

**Comment by MahdiBM** [comment]
<not actually ready for review>

**Comment by MahdiBM** [comment]
@gwynne Does this look _slightly_ better than before?
I removed all the "shared" instances.
This is just 1/3 of the work there .... I'll need to do another round of more cleanup and one last round to simplify the tests probably.

**Comment by MahdiBM** [comment]
@gwynne need https://github.com/gwynne/swift-semver/pull/1 for full `-strict-concurrency=complete` compatibility.
Then i can update the PR.

File: Sources/Penny/+Array.swift [hunk 1]
```diff
@@ -28,6 +28,12 @@ extension Array {
 }
 
 extension Array<String> {
+    mutating func appendUnique(_ element: String) {
```
Comment (gwynne) [inline Sources/Penny/+Array.swift:31]: This is just _screaming_ out that what you really want here is `swift-collections`'s `OrderedSet`

**Review by gwynne — APPROVED** [review]
I'm hitting approve here despite the one nit... I really would be happier if you addressed it, but I'm leaving it up to you.

File: Sources/Penny/+Array.swift (hunk 1 above)
Comment (MahdiBM) [reply Sources/Penny/+Array.swift:31]: Hmmm good idea

**Pull request merged**